app/scripts/controller-init/confirmations/transaction-controller-init.ts

```diff
@@ -153,6 +157,15 @@ export const TransactionControllerInit: ControllerInitFunction<
           transactionMeta,
           rawTx,
         ),
+      publishBatch: async (_request: PublishBatchHookRequest) =>
+        await publishBatchSmartTransactionHook({
```

Comment:
As these `-init` files are to decouple the controllers and simplify code ownership, it would be great in future if we could move "everything" into `lib/smart-transactions` and perhaps just invoke it here via a messenger call such as `SmartTransactions:publishHook`, or at least construct the class directly without all the logic below.

Reply:
Nice idea, will create a ticket for moving STX specific code into lib/smart-transactions.